### PR TITLE
Support gpu version crossvalidator for LinearRegression

### PR DESCRIPTION
**Pull request by wbo4958**

### CODE REVIEW

**Comment by wbo4958** [comment]
build

File: python/src/spark_rapids_ml/regression.py [hunk 1]
```diff
@@ -524,25 +608,30 @@ def _create_pyspark_model(self, result: Row) -> "LinearRegressionModel":
...
+    _RegressionModelEvaluationMixIn,
 ):
     """Model fitted by :class:`LinearRegression`."""
 
     def __init__(
         self,
-        coef_: List[float],
-        intercept_: float,
+        coef_: Union[List[float], List[List[float]]],
+        intercept_: Union[float, List[float]],
```
Comment (leewyang) [inline python/src/spark_rapids_ml/regression.py:626]: Wondering if just assuming lists here would simplify things elsewhere, where the single model is just represented as a list of one item.

**Review by leewyang — APPROVED** [review]
One question/comment, but otherwise LGTM

File: python/src/spark_rapids_ml/regression.py (hunk 1 above)
Comment (wbo4958) [reply python/src/spark_rapids_ml/regression.py:626]: Wow, Good suggestion. I will file a followup PR to handle it not only for LinearRegression but also for RandomForest.

**Pull request merged**